### ISSUE TITLE
CalledProcessError when allowing http auth via dulwich

### ISSUE THREAD

**Issue by fazal-kh4n** [python-poetry/poetry#5573]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS Monterey 12.2.1
- **Poetry version**: Poetry version 1.1.13
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: [Link](https://gist.github.com/fazal-kh4n/ccd3e88d18926fcf883d968096c9243f)

## Issue

I am trying to add a dependency from a private repo following the instructions here:

https://github.com/python-poetry/poetry/pull/5567

The steps I followed:
```
poetry config repositories.git-repo https://gitlhub.com/org/project.git
poetry config http-basic.git-repo fazal-kh4n github_access_token
poetry add git+https://github.com/org/project.git
```

The final step, `poetry add git+https://github.com/org/project.git` however gives me an error:

```
 CalledProcessError
 
  Command '['git', '--git-dir', '/var/folders/4l/1vywq2ds0nb6hhkc1w6_ybjh0000gn/T/pypoetry-git-cc-python-client-authq0unw11z/.git', '--work-tree', '/var/folders/4l/1vywq2ds0nb6hhkc1w6_ybjh0000gn/T/pypoetry-git-cc-python-client-authq0unw11z', 'checkout', 'master']' returned non-zero exit status 1.

  at ~/.poetry/lib/poetry/utils/_compat.py:217 in run
      213│                 process.wait()
      214│                 raise
      215│             retcode = process.poll()
      216│             if check and retcode:
    → 217│                 raise CalledProcessError(
      218│                     retcode, process.args, output=stdout, stderr=stderr
      219│                 )
      220│         finally:
      221│             # None because our context manager __exit__ does not use them.

```

Is this a bug, or have I misunderstood the instructions?

**Side Note:**
I am able to add a private dependency by adding the following line to my pyproject.toml file:

`package-name = {git = 'git+https://GIT_ACCESS_TOKEN@github.com/CroudTech/package.git'}`

But the issue is, I want the GIT_ACCESS_TOKEN to be an env var, not a raw string for security reasons.

**Comment by abn** [comment]
@fazal-kh4n https://github.com/python-poetry/poetry/pull/5567 is only available on `poetry@master`, not `1.1.13`. The called process error I suspect is a fallback or you are running the stable poetry version.

> `package-name = {git = 'git+https://GIT_ACCESS_TOKEN@github.com/CroudTech/package.git'}`

Is not really supported nor is environent variables in dependency specifications.

**Comment by abn** [comment]
You can try the fix at #5567 like shown here.

### Using pipx
```sh
pipx install --suffix=@5567 'poetry @ git+https://github.com/python-poetry/poetry.git@refs/pull/5567/head'
poetry@5567 new foobar
cd foobar
poetry@5567 config repositories.git-repo https://gitlhub.com/org/project.git
poetry@5567 config http-basic.git-repo fazal-kh4n ${GIT_ACCESS_TOKEN}
poetry@5567 add git+https://github.com/org/project.git
```

### Using a container (podman | docker)
```sh
podman run --rm -i -e  --entrypoint bash docker.io/python:3.10 <<EOF
set -xe
python -m pip install -q git+https://github.com/python-poetry/poetry.git@refs/pull/5567/head
poetry new foobar
cd foobar
poetry config repositories.git-repo https://gitlhub.com/org/project.git
poetry config http-basic.git-repo fazal-kh4n ${GIT_ACCESS_TOKEN}
poetry add git+https://github.com/org/project.git
EOF
```

**Comment by abn** [comment]
Also note that your configuration has a typo.

```diff
- poetry config repositories.git-repo https://gitlhub.com/org/project.git
+ poetry config repositories.git-repo https://github.com/org/project.git
```
